backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/cardbutton/internal/BpkCardButtonImpl.kt

```diff
@@ -0,0 +1,151 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.cardbutton.internal
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.spring
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.selection.toggleable
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.semantics.Role
+import net.skyscanner.backpack.compose.cardbutton.BpkCardButtonSize
+import net.skyscanner.backpack.compose.cardbutton.BpkCardButtonStyle
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.icon.BpkIconSize
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.tokens.Heart
+import net.skyscanner.backpack.compose.tokens.HeartOutline
+import net.skyscanner.backpack.compose.tokens.ShareIos
+import net.skyscanner.backpack.compose.utils.clickable
+
+internal enum class BpkCardButtonState {
+  Default,
+  Transition
+}
+
+@Composable
+fun BpkSaveCardButtonImpl(
+  checked: Boolean,
+  contentDescription: String,
+  style: BpkCardButtonStyle,
+  size: BpkCardButtonSize,
+  onCheckedChange: (Boolean) -> Unit
+) {
+  var state by remember { mutableStateOf(BpkCardButtonState.Default) }
+  val scaleAnimation by animateFloatAsState(
```

Comment:
the animation looks a little different to the spec in figma - it might be worth sitting with adam to tweak this

Reply:
spent some time and tweaked it. now it looks more similar IMO.

Reply:
looking at the spec in figma it's still a bit different (the spec is written out on the right in figma)